walker.py

```diff
@@ -0,0 +1,138 @@
+                                    size=(number_of_steps, 2))
+        scene.append(walker)
+
+    return scene
+
+  
+def calculate_the_path(walker, number_of_steps):
+    """
+    Calculates the path the walker walks
+    :param walker: The array representing the walker, holding random coordinates
+    :param number_of_steps: the number of steps the walker takes
+    :return: the updated array representing the walker, holding all coordinates the walker walks to
+    """
+    # split coordinates
+    x_coord = walker[:, 0]
+    y_coord = walker[:, 1]
+
+    # calculate new coordinates for each step
+    for stepnumber in range(1, number_of_steps):
+        direction_of_step = np.random.randint(1, 4)
+        if direction_of_step == 1:      # east
+            x_coord[stepnumber] = x_coord[stepnumber - 1] + 1
+            y_coord[stepnumber] = y_coord[stepnumber - 1]
+        elif direction_of_step == 2:    # west
+            x_coord[stepnumber] = x_coord[stepnumber - 1] - 1
+            y_coord[stepnumber] = y_coord[stepnumber - 1]
+        elif direction_of_step == 3:    # north
+            x_coord[stepnumber] = x_coord[stepnumber - 1]
+            y_coord[stepnumber] = y_coord[stepnumber - 1] + 1
+        else:                           # south
+            x_coord[stepnumber] = x_coord[stepnumber - 1]
+            y_coord[stepnumber] = y_coord[stepnumber - 1] - 1
+
+    # return all coordinates of the path
+    return walker
+
+
+def plot_the_paths(list_of_walkers, outputfilename):
+    """
+    Creates the plot of the calculated path of the walker
+    :param list_of_walkers: Arrays representing the walkers. Each array holds the
+    coordinates of the respective path the walker walked.
+    :param outputfilename: The file which will be created with the plotted paths
+    """
+    # define the number of rows and columns of subplots
+    if len(list_of_walkers) == 1:
+        columns_of_plots = 1
+    else:
+        columns_of_plots = 2
+    rows_of_plots = round(len(list_of_walkers)/2)
+
+    # create the subplots
+    figure, axis = plt.subplots(rows_of_plots, columns_of_plots)
+    for counter in enumerate(list_of_walkers):
+        # get the coordinates of this walker, get its subplot and plot the path in it
+        x_coords = counter[1][:, 0]
+        y_coords = counter[1][:, 1]
+
+        row = int(counter[0]/2)
+        if counter[0] % 2 == 0:
+            column = 0
+        else:
+            column = 1
+
+        axis[row, column].plot(x_coords, y_coords)
+        axis[row, column].set_title(f'Walker {counter[0]+1}')
+
+    if len(list_of_walkers) % 2 != 0:
+        # if the number of walkers is odd, delete the last (unused) subplot
+        figure.delaxes(axis[(rows_of_plots - 1), 1])
+
+    # arange subplot to not interfere each other, save the figure to the outputfile and
+    # show the plot to the user
+    figure.tight_layout()
+    plt.savefig(outputfilename)
+    plt.show()
+
+
+def main():
```

Comment:
Readable and understandable! Good job!

Reply:
Think about the requirement of adding walkers with different behavior. It might be difficult to extend your code to include this.

Reply:
Thank you!

Reply:
> Think about the requirement of adding walkers with different behavior. It might be difficult to extend your code to include this.

Yes, you're right... but we were running out of time yesterday. But we'll think about how to adjust it for the upcoming tasks